src/main.rs

```diff
@@ -0,0 +1,113 @@
+use clap::Parser;
+use serde_json::{Result, Value};
+use std::{fs::File, io::BufReader};
+
+#[derive(Parser, Debug)]
+#[command(version, about, long_about = None)]
+struct Args {
+    #[arg(short, long)]
+    file: String,
+
+    #[arg(short, long)]
+    output: String,
+
+    #[arg(short, long, action)]
+    pretty: bool,
+}
+
+fn main() {
+    env_logger::init();
+    let args = Args::parse();
+
+    log::debug!(
+        "Attempting read from {}, and will try write to {}",
+        args.file,
+        args.output
+    );
+
+    let json_data = &mut read_from_file(args.file).expect("Failed to parse the file contents!");
+
+    remove_null_values(json_data);
+    write_to_file(args.output, json_data, args.pretty);
+
+    println!("Successfully processed file.")
+}
+
+fn read_from_file(file_path: String) -> Result<Value> {
+    let file = File::open(file_path).expect("Could not open file!");
+    let reader = BufReader::new(file);
+    let v = serde_json::from_reader(reader)?;
+
+    Ok(v)
+}
+
+fn write_to_file(path: String, json_data: &mut Value, should_pretty_print: bool) {
+    let file = &mut File::create(path).expect("Could not create file!");
+
+    log::debug!("Should pretty print: {}", should_pretty_print);
+
+    if should_pretty_print {
+        serde_json::to_writer_pretty(file, json_data).expect("Could not write to file");
+    } else {
+        serde_json::to_writer(file, json_data).expect("Could not write to file");
+    }
+}
+
+fn remove_null_values(json_data: &mut Value) {
+    match json_data {
+        Value::Object(_) => remove_null_values_from_object(json_data),
+        Value::Array(_) => remove_null_values_from_array(json_data),
+        _ => (),
+    }
+}
+
+fn remove_null_values_from_object(json_data: &mut Value) {
```

Comment:
As stated above, don't take `&mut Value`, take `&mut Map<String, Value>` the contents of `Value::Object`

Reply:
Nooow we're talking! This is great, I was a bit confused that I couldn't specify the type as `&mut Value::Object`, but this makes me understand what was going on. Thanks!

Reply:
Wow, this cascades and helps out quite a lot

Reply:
Yes, it's a bit unfortunate you can't use `Value::Object` as type. As a result you'll often see something like
```
struct Bar;
struct Baz;
enum Foo {
    Bar(Bar),
    Baz(Baz),
}
```
rather than struct variants so you can use the variants independent of the enum.